gateway/gateway.go

```diff
@@ -282,54 +323,85 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 	// errors will occur.
 	// Unexpected behavior will also occur if stopGateway() is called when the gateway hasn't been
 	// started, so only call stopGateway if this module already started the gateway.
-	if g.started {
-		g.reset(ctx)
+	if err := g.resetConcentrators(ctx); err != nil {
+		return err
 	}
 
 	// workers and process are restarted during reconfigure
-	g.closeProcess()
-	if g.workers != nil {
-		g.workers.Stop()
+	g.closeProcesses()
+
+	// maintain devices and lastReadings through reconfigure.
+	if g.devices == nil {
+		g.devices = make(map[string]*node.Node)
 	}
 
-	cfg, err := getNativeConfig(conf)
-	if err != nil {
-		return err
+	if g.lastReadings == nil {
+		g.lastReadings = make(map[string]interface{})
 	}
 
+	g.concentrators = make([]*concentrator, 0)
+
+	switch conf.Model {
+	case ModelRak7391:
```

Comment:
kinda wish there was another way for us to check if its a multi config outside of checking all of the model names here. not too sure what that would look like though